pkg/cmd/server/kubernetes/network/network.go

```diff
@@ -232,17 +237,37 @@ func (c *NetworkConfig) RunProxy() {
 }
 
 // getNodeIP is copied from the upstream proxy config to retrieve the IP of a node.
-func getNodeIP(client kv1core.CoreV1Interface, hostname string) net.IP {
-	var nodeIP net.IP
-	node, err := client.Nodes().Get(hostname, metav1.GetOptions{})
-	if err != nil {
-		glog.Warningf("Failed to retrieve node info: %v", err)
-		return nil
+func getNodeIP(client kv1core.CoreV1Interface, hostname string) (net.IP, error) {
+	var node *v1.Node
+	var nodeErr error
+
+	// We may beat the thread that causes the node object to be created,
+	// so if we can't get it, then we need to wait.
+	// This will wait 0, 2, 4, 8, ... 256 seconds, for a total of 8.5 mins
```

Comment:
Do we really want to wait for ~8 mins (seems a bit long)?
In openshift SDN node, we wait for local subnet which requires node registration and master reacting with this event and creating hostSubnet. Total wait timeout in this case is only 1 min (actually 30 secs because of the bug), this may be low. We did not see issues with this, to be on safer side (when node loaded) 2 to 3 mins may suffice.

Reply:
My testing showed registration happened by the 2nd or 3rd attempt... so 2-4 seconds.  I'll lower it.